File: Scripts/results/generated_code_results/Flutter/bak_code/Social Networking/SocialSoftware/UserProfileScreen.dart
import 'package:flutter/material.dart';
class UserProfileScreen extends StatefulWidget {
@override
_UserProfileScreenState createState() => _UserProfileScreenState();
}
class _UserProfileScreenState extends State<UserProfileScreen> with SingleTickerProviderStateMixin {
late TabController _tabController;
bool _isFollowing = false;
@override
void initState() {
super.initState();
_tabController = TabController(length: 2, vsync: this);
}
@override
void dispose() {
_tabController.dispose();
super.dispose();
}
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
leading: IconButton(
icon: Icon(Icons.arrow_back),
onPressed: () => Navigator.pop(context),
),
title: Text('Voltar'),
actions: [
IconButton(
icon: Icon(Icons.link),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.more_vert),
onPressed: () {},
),
],
),
body: ListView(
children: [
Padding(
padding: const EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Row(
mainAxisAlignment: MainAxisAlignment.spaceBetween,
children: [
Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text(
'Alisson Rochinski',
style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),
),
Text('@arochinski'),
Chip(label: Text('threads.net')),
Text("UI Designer | Let's redesign the world"),
Text('alisson.rochinski@gmail.com'),
SizedBox(height: 8),
Row(
children: [
Stack(
children: List.generate(3, (index) => Positioned(
left: index * 20.0,
child: CircleAvatar(
radius: 12,
backgroundImage: AssetImage('assets/images/template.png'),
),
)),
),
SizedBox(width: 36),
Text('2957 seguidores'),
],
),
],
),
Stack(
children: [
CircleAvatar(
radius: 40,
backgroundImage: AssetImage('assets/images/template.png'),
),
Positioned(
right: 0,
bottom: 0,
child: Icon(Icons.verified, color: Colors.blue),
),
],
),
],
),
SizedBox(height: 16),
ElevatedButton(
onPressed: () {
setState(() {
_isFollowing = !_isFollowing;
});
},
child: Text(_isFollowing ? 'Following' : 'Follow'),
style: ElevatedButton.styleFrom(
minimumSize: Size(double.infinity, 40),
),
),
],
),
),
TabBar(
controller: _tabController,
tabs: [
Tab(text: 'Chirps'),
Tab(text: 'Answers'),
],
),
Container(
height: 300, // Adjust as needed
child: TabBarView(
controller: _tabController,
children: [
_buildChirpsList(),
Center(child: Text('No answers yet')),
],
),
),
],
),
bottomNavigationBar: BottomNavigationBar(
type: BottomNavigationBarType.fixed,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),
BottomNavigationBarItem(icon: Icon(Icons.search), label: 'Search'),
BottomNavigationBarItem(icon: Icon(Icons.add_box), label: 'Create'),
BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Likes'),
BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),
],
onTap: (index) {
switch (index) {
case 0:
Navigator.pushNamed(context, 'Feed');
break;
case 1:
Navigator.pushNamed(context, 'Search');
break;
case 3:
Navigator.pushNamed(context, 'Activity');
break;
}
},
),
);
}
Widget _buildChirpsList() {
return ListView.builder(
itemCount: 1,
itemBuilder: (context, index) {
return ListTile(
leading: CircleAvatar(
backgroundImage: AssetImage('assets/images/template.png'),
),
title: Text('arochinski'),
subtitle: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
Text('Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint'),
SizedBox(height: 8),
Row(
children: [
IconButton(
icon: Icon(Icons.favorite_border),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.chat_bubble_outline),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.repeat),
onPressed: () {},
),
IconButton(
icon: Icon(Icons.send),
onPressed: () {},
),
],
),
Text('7 answers · 59 likes'),
],
),
trailing: Text('2 min'),
);
},
);
}
}

File: Scripts/results/generated_code_results/ablation_study/sam/Social Networking/SocialSoftware/UserProfileScreen.js
import React from 'react';

import { View, Text, Image, StyleSheet, TouchableOpacity, ScrollView, FlatList } from 'react-native';

import { useNavigation } from '@react-navigation/native';

import Icon from 'react-native-vector-icons/Ionicons';



const UserProfileScreen = ({ route }) => {

  const navigation = useNavigation();

  const { username } = route.params;



  const userInfo = {

    name: 'Alisson Rochinski',

    username: 'arochinski',

    website: 'threads.net',

    bio: "UI Designer | Let's redesign the world",

    email: 'alisson.rochinski@gmail.com',

    followers: 2957,

  };



  const posts = [

    {

      content: 'Tempor dolor proident duis incididunt et. Cillum incididunt irure nisi excepteur sit ad ipsum consectetur proident deserunt sint',

      stats: { answers: 7, likes: 59 },

      timeAgo: '2 min',

    },

  ];



  const renderPost = ({ item }) => (

    <View style={styles.post}>

      <Text style={styles.postContent}>{item.content}</Text>

      <View style={styles.postStats}>

        <Text style={styles.statText}>{item.stats.answers} answers</Text>

        <Text style={styles.statText}>{item.stats.likes} likes</Text>

        <Text style={styles.timeAgo}>{item.timeAgo}</Text>

      </View>

    </View>

  );



  return (

    <View style={styles.container}>

      <View style={styles.header}>

        <TouchableOpacity onPress={() => navigation.goBack()}>

          <Icon name="arrow-back" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity>

          <Icon name="logo-instagram" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity>

          <Icon name="ellipsis-horizontal" size={24} color="#fff" />

        </TouchableOpacity>

      </View>



      <ScrollView>

        <View style={styles.profileInfo}>

          <Image source={require('../assets/snack-icon.png')} style={styles.profilePicture} />

          <Text style={styles.name}>{userInfo.name}</Text>

          <Text style={styles.username}>{userInfo.username}</Text>

          <Text style={styles.bio}>{userInfo.bio}</Text>

          <Text style={styles.website}>{userInfo.website}</Text>

          <Text style={styles.followers}>{userInfo.followers} followers</Text>

        </View>



        <TouchableOpacity style={styles.followButton}>

          <Text style={styles.followButtonText}>Follow</Text>

        </TouchableOpacity>



        <View style={styles.tabs}>

          <TouchableOpacity style={styles.tab}>

            <Text style={styles.tabText}>Chirps</Text>

          </TouchableOpacity>

          <TouchableOpacity style={styles.tab}>

            <Text style={styles.tabText}>Answers</Text>

          </TouchableOpacity>

        </View>



        <FlatList

          data={posts}

          renderItem={renderPost}

          keyExtractor={(item, index) => index.toString()}

        />

      </ScrollView>



      <View style={styles.bottomNav}>

        <TouchableOpacity onPress={() => navigation.navigate('Feed')}>

          <Icon name="home-outline" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Search')}>

          <Icon name="search-outline" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity>

          <Icon name="add-circle-outline" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Activity')}>

          <Icon name="heart-outline" size={24} color="#fff" />

        </TouchableOpacity>

        <TouchableOpacity onPress={() => navigation.navigate('Profile')}>

          <Icon name="person-outline" size={24} color="#fff" />

        </TouchableOpacity>

      </View>

    </View>

  );

};



const styles = StyleSheet.create({

  container: {

    flex: 1,

    backgroundColor: '#000',

  },

  header: {

    flexDirection: 'row',

    justifyContent: 'space-between',

    alignItems: 'center',

    padding: 16,

    borderBottomWidth: 1,

    borderBottomColor: '#333',

  },

  profileInfo: {

    alignItems: 'center',

    padding: 16,

  },

  profilePicture: {

    width: 80,

    height: 80,

    borderRadius: 40,

    marginBottom: 8,

  },

  name: {

    fontSize: 18,

    fontWeight: 'bold',

    color: '#fff',

  },

  username: {

    fontSize: 16,

    color: '#999',

    marginBottom: 8,

  },

  bio: {

    fontSize: 14,

    color: '#fff',

    textAlign: 'center',

    marginBottom: 8,

  },

  website: {

    fontSize: 14,

    color: '#3897f0',

    marginBottom: 8,

  },

  followers: {

    fontSize: 14,

    color: '#999',

  },

  followButton: {

    backgroundColor: '#fff',

    paddingVertical: 8,

    paddingHorizontal: 16,

    borderRadius: 20,

    alignSelf: 'center',

    marginVertical: 16,

  },

  followButtonText: {

    color: '#000',

    fontWeight: 'bold',

  },

  tabs: {

    flexDirection: 'row',

    borderBottomWidth: 1,

    borderBottomColor: '#333',

  },

  tab: {

    flex: 1,

    alignItems: 'center',

    paddingVertical: 16,

  },

  tabText: {

    color: '#fff',

    fontWeight: 'bold',

  },

  post: {

    padding: 16,

    borderBottomWidth: 1,

    borderBottomColor: '#333',

  },

  postContent: {

    color: '#fff',

    fontSize: 14,

    marginBottom: 8,

  },

  postStats: {

    flexDirection: 'row',

    justifyContent: 'space-between',

  },

  statText: {

    color: '#999',

    fontSize: 12,

  },

  timeAgo: {

    color: '#999',

    fontSize: 12,

  },

  bottomNav: {

    flexDirection: 'row',

    justifyContent: 'space-around',

    alignItems: 'center',

    paddingVertical: 12,

    borderTopWidth: 1,

    borderTopColor: '#333',

  },

});



export default UserProfileScreen;